src/sonic-config-engine/minigraph.py

```diff
@@ -369,7 +456,7 @@ def parse_cpg(cpg, hname):
                         'keepalive': keepalive,
                         'nhopself': nhopself
                     }
-                else:
+                elif start_router.lower() == hname.lower():
```

Comment:
why do we need it here?

Reply:
in case of multiasic platform minigraph, Cpg will have all asic and device BGP session information. While parsing for specific asic, we have to make sure either start_router or end_router matches the name of the asic of interest.